### DISCUSSION TITLE
Oracle Connection String bei Suche/Auswahlwerteliste

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/webgis-community#75] [Q&A]
Hallo Jürgen!

Ich möchte auf unsere WIS Oracle-DB zugreifen, um bei der Suche eine Auswahlwerteliste angeboten zu bekommen. Leider funktioniert die Auswahlwerteliste nicht. 
In der webgis_exceptions.log-Datei erscheint folgende Fehlermeldung:

`QUERY (Select distinct TYP from *** order by TYP): 'SERVER' ist ein ungültiges Verbindungszeichenfolge-Attribut`

Der Connection String ist im Moment folgendermaßen eingetragen:

`ORACLE:SERVER=(DESCRIPTION=(ADDRESS_LIST=(ADDRESS=(PROTOCOL=TCP)(HOST=***.***.at)(PORT=***)))(CONNECT_DATA=(SERVER=DEDICATED)(SERVICE_NAME=***.***.at)));User Id=***;Password=***;`

Der Connection String ist gleich wie im webgis4. Dort funktioniert es.
Muss ich im webgis5 (5.23.2102) beim Connection String etwas anpassen/ändern und deutet die Fehlermeldung auf einen anderen Fehler hin?

Danke und einen schönen Dienstag!

Lg, Andi

**Comment by gis-eni** [comment]
Hallo Andi,

Die Verbindung zur Oracle erfolgt mit Oracle.ManagedADataAccess.Core Library von Oracle. Die hat den Vorteil, dass nicht extra einen Oracle Client installieren muss. 
Allerdings schauen damit die ConnectionStrings unter umständen anders aus, als mit alten Libraries. Probier einmal anstatt ``SERVER=`` das eher aktuellere ``Data Source=``. Also zum Beispiel:
``User Id=myUser;Password=myPassword;Data Source=MyOracleConnection``

Ich hab leider leider keine Oracle Datenbank, um das zu testen...

**Reply by Andreas-Lampert** [reply]
Hallo Jürgen!

Danke für die Info! 
Der Connection String muss mit **Data Source=** eingetragen werden, dann funktioniert es. 

`Oracle:Data Source=(DESCRIPTION=(ADDRESS_LIST=(ADDRESS=(PROTOCOL=TCP)(HOST=***.***.at)(PORT=***)))(CONNECT_DATA=(SERVER=DEDICATED)(SERVICE_NAME=***.***.at)));User Id=***;Password=***;`
